torch/_dynamo/utils.py

```diff
@@ -1715,3 +1716,176 @@ def build_checkpoint_variable(**options):
         activation_checkpoint_op,
         **options,
     )
+
+
+# The following functions are adapted from
+# https://github.com/python/cpython/blob/v3.11.4/Lib/traceback.py
+# in order to output source code corresponding to bytecode in 3.11+.
+# We need our own versions since we want to support multiline expressions.
```

Comment:
How is it adapted vs the same? Should I diff with traceback.py?

Reply:
I used the functions in traceback.py as a starting point. The logical structure of the code is similar, but there are enough changes so that we shouldn't consider these functions the same.